src/components/buttons/HdLoaderButton.vue

```diff
@@ -76,15 +109,23 @@ export default {
       type: Number,
       default: 4,
     },
-  },
-  mounted() {
-    this.undrawAll();
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
   },
   data() {
     return {
       state: 'idle',
```

Comment:
I would take the advantage to make an "Enum" out of it. There are a bunch of String comparisons happening and it might get lost in the future.